Makefile

```diff
@@ -26,23 +26,42 @@ SYSLIBDIR ?= /lib
 DATADIR ?= /usr/share
 PA_VER_FULL ?= $(shell pkg-config --modversion libpulse | cut -d "-" -f 1 || echo 0.0)
 PA_MODULE_DIR ?= $(shell pkg-config --variable=modlibexecdir libpulse)
+VCHAN_LIBS := $(shell pkg-config --libs vchan)
+VCHAN_CFLAGS := $(shell pkg-config --cflags vchan)
+XORG_SERVER_CFLAGS=$(shell pkg-config --cflags xorg-server)
 
 help:
 	@echo "Qubes GUI main Makefile:" ;\
 	    echo; \
 	    echo "make clean                <--- clean all the binary files";\
-	    @exit 0;
+	    exit 0;
 
 appvm: gui-agent/qubes-gui gui-common/qubes-gui-runuser \
 	xf86-input-mfndev/src/.libs/qubes_drv.so \
 	xf86-video-dummy/src/.libs/dummyqbs_drv.so pulse/module-vchan-sink.so \
 	xf86-qubes-common/libxf86-qubes-common.so
 
-gui-agent/qubes-gui:
-	$(MAKE) -C gui-agent
+OBJS = vmside.o \
+	../gui-common/txrx-vchan.o ../gui-common/error.o ../common/list.o ../gui-common/encoding.o
```

Comment:
I don't like moving compilation of everything in `gui-agent/` into top level makefile. For example the above paths look misleading now, but also you pollute env for building other parts (xorg driver, pulseaudio etc) with agent-specific settings. You workaround some of it by using non-standard variable names, but that isn't a good idea.
But, since `gui-common/*` and `common/*`  is in fact used only in `gui-agent/` (not really shared with anything else), maybe simply move them all into `gui-agent/` dir?

Reply:
This still applies.